src/gdrive.js

```diff
@@ -0,0 +1,287 @@
+import _ from 'underscore'
+import fs from 'fs'
+import google from 'googleapis'
+import googleAuth from 'google-auth-library'
+
+let GDrive = function () {
+
+  let PROMISES = 3
+  let DRIVE_ROOT = process.env.DRIVE_ROOT
+  const SERVICE_CREDENTIALS = JSON.parse(process.env.SERVICE_CREDENTIALS)
+  const SCOPES_DRIVE = ['https://www.googleapis.com/auth/drive.metadata.readonly']
+  const SCOPES_SHEETS = ['https://www.googleapis.com/auth/spreadsheets']
+  let FILES = []
+  let FOLDERS = []
+
+  /**
+   * Get the array of json objects containing all documents found in the
+   * library
+   *
+   * @param driveRoot Optional: ID of the root folder to traverse
+   * @param promises Optional: Number of parallel promises per iteration
+   *   (default
+   *   3)
+   * @returns {Promise} Promise resolves an object {files: [...], paths: [...]}
+   */
+  function getGDriveDocuments(driveRoot = '', promises = '') {
+    if (driveRoot != '') {
+      DRIVE_ROOT = driveRoot
+    }
+    if (promises != '') {
+      PROMISES = promises
```

Comment:
@dumyan can you explain how this PROMISES works? What problem are you solving here.